### PR TITLE
Update Configuration Docs

### PR DESCRIPTION
**Pull request by eslickj**
This brings back the configuration doc page, and makes a few minor changes that let you make changes to the configuration on the fly a bit easier.  Also contains docs for configuring "ipopt-idaes" solver.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/user_guide/configuration.rst
Comment (blnicho) [inline docs/user_guide/configuration.rst:126]: Typo: "apply **any** resulting"

File: docs/user_guide/configuration.rst
Comment (blnicho) [inline docs/user_guide/configuration.rst:140]: I think you should list the default configuration setting for all of these entries

File: docs/user_guide/configuration.rst
Comment (blnicho) [inline docs/user_guide/configuration.rst:217]: Typo: "for ``ipopt`` to **use** idaes-provided"

File: docs/user_guide/configuration.rst
Comment (blnicho) [inline docs/user_guide/configuration.rst:214]: I think you should split this into two sentences: This is a config block that provides the default configuration for the ``ipopt-idaes`` solver. ``ipopt-idaes`` is a wrapper for the Pyomo ``ipopt`` solver class that allows default solver options to be configured in the general idaes ConfigBlock.

File: idaes/__init__.py
Comment (blnicho) [inline idaes/__init__.py:19]: Typo: directories

File: idaes/config.py
Comment (blnicho) [inline idaes/config.py:35]: Typo: known

File: idaes/config.py
Comment (blnicho) [inline idaes/config.py:240]: Typo: accessible

**Comment by eslickj** [comment]
Typos should be fixed.

File: idaes/config.py
Comment (jsiirola) [inline idaes/config.py:152]: FWIW, you are inconsistent with your use of `ConfigBlock` (the old name) and `ConfigDict` (the new name).  They are the same class, and we should probably just pick one for using everywhere.

File: idaes/config.py
Comment (jsiirola) [inline idaes/config.py:145]: This is not really necessary.  You can always do `config.value()`, which returns the data structure with `ConfigDict` resolved to `dict`, `ConfigList` to `list, and `ConfigValue` to the contained value.  You would still need to map `set()` -> `list`, though.

We could also add a JSON serializer to the ConfigBlock system (analogous to the YAML template generator)

File: docs/user_guide/configuration.rst
Comment (jsiirola) [inline docs/user_guide/configuration.rst:145]: This shouldn't hold up this PR, but one of the driving motivations behind the Config system was to include documentation *in* the configuration.  So, we *could* move this into the Config definitions and then autogenerate this text here.

**Review by jsiirola — DISMISSED** [review]
Overall, this looks good.  Several comments that you can decide to either tackle here or defer to another PR some other time.

**Review by blnicho — DISMISSED** [review]

File: idaes/config.py
Comment (eslickj) [reply idaes/config.py:152]: Yeah.  I noticed that.  Then I started to worry that ConfigBlock would be deprecated and couldn't decide what to do.  Started to think I better use ConfigDict, but didn't really follow through.  I guess I'll go with ConfigBlock for now since that's the way it is in all our models.

File: idaes/config.py
Comment (eslickj) [reply idaes/config.py:145]: Thanks.  I knew that much be there, but I didn't immediately figure out how to do it.  I'll update.

I think a json serialized would be a nice addition if you have time.

File: docs/user_guide/configuration.rst
Comment (eslickj) [reply docs/user_guide/configuration.rst:145]: Yeah.  That's a good point, and I should have thought of that because we document our models that way.  I'll go ahead and switch this over if it's not too hard.

**Comment by eslickj** [comment]
@jsiirola and @blnicho this should be good to go again.  I just made some really minor changes to address John's first two comments since you approved it.  I couldn't immediately figured out an easy way to generate sphinx docs from the ConfigBlock object yet, so I'm going to let that go for now.

**Review by blnicho — APPROVED** [review]

**Review by jsiirola — APPROVED** [review]

**Pull request merged**
